pkg/asset/agent/image/agentartifacts.go

```diff
@@ -61,8 +65,10 @@ func (a *AgentArtifacts) Generate(_ context.Context, dependencies asset.Parents)
 	agentClusterInstall := &manifests.AgentClusterInstall{}
 	registriesConf := &mirror.RegistriesConf{}
 	agentconfig := &config.AgentConfig{}
+	agentWorkflow := &workflow.AgentWorkflow{}
+	addNodesConfig := &joiner.AddNodesConfig{}
 
-	dependencies.Get(ignition, kargs, baseIso, agentManifests, agentClusterInstall, registriesConf, agentconfig)
+	dependencies.Get(ignition, kargs, baseIso, agentManifests, agentClusterInstall, registriesConf, agentconfig, agentWorkflow, addNodesConfig)
```

Comment:
nit: to slightly increase the readability, the assets that are required only for day2 could be get later within the case, ie :

```
...
case workflow.AgentWorkflowTypeAddNodes:
        agentWorkflow := &workflow.AgentWorkflow{}
	addNodesConfig := &joiner.AddNodesConfig{}
        dependencies.Get(agentWorkflow, addNodesConfig)
...
```